Flakes in `uv publish` integration test

```
Run ./uv run -p 3.12 scripts/publish/test_publish.py --uv ./uv all
DEBUG uv 0.4.24 (4d2f31cb7 2024-10-21)
Reading inline script metadata from `scripts/publish/test_publish.py`
DEBUG Searching for Python 3.12 in managed installations or system path
DEBUG Searching for managed installations at `/home/runner/.local/share/uv/python`
DEBUG Found `cpython-3.12.7-linux-x86_64-gnu` at `/opt/hostedtoolcache/Python/3.12.7/x64/bin/python` (search path)
DEBUG Caching via interpreter: `/opt/hostedtoolcache/Python/3.12.7/x64/bin/python`
DEBUG Using request timeout of 30s
DEBUG Solving with installed Python version: 3.12.7
DEBUG Solving with target Python version: >=3.12.7
DEBUG Adding direct dependency: httpx>=0.27.2, <0.28
DEBUG Adding direct dependency: packaging>=24.1, <25
DEBUG No cache entry for: https://pypi.org/simple/httpx/
DEBUG No cache entry for: https://pypi.org/simple/packaging/
DEBUG Searching for a compatible version of httpx (>=0.27.2, <0.28)
DEBUG Selecting: httpx==0.27.2 [compatible] (httpx-0.27.2-py3-none-any.whl)
DEBUG No cache entry for: https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb[23](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:24)3fe0416e4639146c089643407ce6/packaging-[24](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:25).1-py3-none-any.whl.metadata
DEBUG No cache entry for: https://files.pythonhosted.org/packages/56/95/9377bcb415797e44274b51d46e3249eba641711cf3348050f76ee7b15ffc/httpx-0.27.2-py3-none-any.whl.metadata
DEBUG Adding transitive dependency for httpx==0.27.2: anyio*
DEBUG Adding transitive dependency for httpx==0.27.2: certifi*
DEBUG Adding transitive dependency for httpx==0.27.2: httpcore>=1.dev0, <2.dev0
DEBUG Adding transitive dependency for httpx==0.27.2: idna*
DEBUG Adding transitive dependency for httpx==0.27.2: sniffio*
DEBUG Searching for a compatible version of packaging (>=24.1, <[25](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:26))
DEBUG Selecting: packaging==24.1 [compatible] (packaging-24.1-py3-none-any.whl)
DEBUG No cache entry for: https://pypi.org/simple/anyio/
DEBUG No cache entry for: https://pypi.org/simple/certifi/
DEBUG No cache entry for: https://pypi.org/simple/httpcore/
DEBUG No cache entry for: https://pypi.org/simple/idna/
DEBUG No cache entry for: https://pypi.org/simple/sniffio/
DEBUG Searching for a compatible version of anyio (*)
DEBUG Selecting: anyio==4.6.2.post1 [compatible] (anyio-4.6.2.post1-py3-none-any.whl)
DEBUG No cache entry for: https://files.pythonhosted.org/packages/e9/44/75a9c9421471a6c4805dbf2356f7c181a29c1879239abab1ea2cc8f38b40/sniffio-1.3.1-py3-none-any.whl.metadata
DEBUG No cache entry for: https://files.pythonhosted.org/packages/12/90/3c9ff0512038035f59d279fddeb79f5f1eccd8859f06d6163c58798b9487/certifi-2024.8.30-py3-none-any.whl.metadata
DEBUG No cache entry for: https://files.pythonhosted.org/packages/06/89/b161908e2f51be56568184aeb4a880fd287178d176fd1c860d2217f41106/httpcore-1.0.6-py3-none-any.whl.metadata
DEBUG No cache entry for: https://files.pythonhosted.org/packages/e4/f5/f2b75d2fc6f1a[26](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:27)0f340f0e7c6a060f4dd2961cc16884ed851b0d18da06a/anyio-4.6.2.post1-py3-none-any.whl.metadata
DEBUG No cache entry for: https://files.pythonhosted.org/packages/76/c6/c88e154df9c4e1a2a66ccf0005a88dfb2650c1dffb6f5ce603dfbd452ce3/idna-3.10-py3-none-any.whl.metadata
DEBUG Adding transitive dependency for anyio==4.6.2.post1: idna>=2.8
DEBUG Adding transitive dependency for anyio==4.6.2.post1: sniffio>=1.1
DEBUG Searching for a compatible version of certifi (*)
DEBUG Selecting: certifi==2024.8.30 [compatible] (certifi-2024.8.30-py3-none-any.whl)
DEBUG Searching for a compatible version of httpcore (>=1.dev0, <2.dev0)
DEBUG Selecting: httpcore==1.0.6 [compatible] (httpcore-1.0.6-py3-none-any.whl)
DEBUG Adding transitive dependency for httpcore==1.0.6: certifi*
DEBUG Adding transitive dependency for httpcore==1.0.6: h11>=0.13, <0.15
DEBUG Searching for a compatible version of idna (>=2.8)
DEBUG Selecting: idna==3.10 [compatible] (idna-3.10-py3-none-any.whl)
DEBUG No cache entry for: https://pypi.org/simple/h11/
DEBUG Searching for a compatible version of sniffio (>=1.1)
DEBUG Selecting: sniffio==1.3.1 [compatible] (sniffio-1.3.1-py3-none-any.whl)
DEBUG Searching for a compatible version of h11 (>=0.13, <0.15)
DEBUG Selecting: h11==0.14.0 [compatible] (h11-0.14.0-py3-none-any.whl)
DEBUG No cache entry for: https://files.pythonhosted.org/packages/95/04/ff642e65ad6b90db43e668d70ffb6736436c7ce41fcc549f4e94722341[27](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:28)/h11-0.14.0-py3-none-any.whl.metadata
DEBUG Tried 8 versions: anyio 1, certifi 1, h11 1, httpcore 1, httpx 1, idna 1, packaging 1, sniffio 1
DEBUG Split specific environment resolution took 0.149s
DEBUG Ignoring empty directory
DEBUG Using request timeout of 30s
DEBUG Identified uncached distribution: anyio==4.6.2.post1
DEBUG Identified uncached distribution: certifi==2024.8.30
DEBUG Identified uncached distribution: h11==0.14.0
DEBUG Identified uncached distribution: httpcore==1.0.6
DEBUG Identified uncached distribution: httpx==0.27.2
DEBUG Identified uncached distribution: idna==3.10
DEBUG Identified uncached distribution: packaging==24.1
DEBUG Identified uncached distribution: sniffio==1.3.1
DEBUG No cache entry for: https://files.pythonhosted.org/packages/12/90/3c9ff0512038035f59d279fddeb79f5f1eccd8859f06d6163c58798b9487/certifi-2024.8.30-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/e4/f5/f2b75d2fc6f1a260f340f0e7c6a060f4dd2961cc16884ed851b0d18da06a/anyio-4.6.2.post1-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/06/89/b161908e2f51be56568184aeb4a880fd[28](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:29)7178d176fd1c860d2217f41106/httpcore-1.0.6-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/56/95/9377bcb415797e44274b51d46e3249eba641711cf3348050f76ee7b15ffc/httpx-0.27.2-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/76/c6/c88e154df9c4e1a2a66ccf0005a88dfb2650c1dffb6f5ce603dfbd452ce3/idna-3.10-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/95/04/ff642e65ad6b90db43e668d70ffb6736436c7ce41fcc549f4e9472234127/h11-0.14.0-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl
DEBUG No cache entry for: https://files.pythonhosted.org/packages/e9/44/75a9c9421471a6c4805dbf2356f7c181a[29](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:30)c1879239abab1ea2cc8f38b40/sniffio-1.3.1-py3-none-any.whl
Installed 8 packages in 18ms
DEBUG Using Python 3.12.7 interpreter at: /home/runner/.cache/uv/archive-v0/qzAMDq6g3xW3blrB8i8AY/bin/python3
DEBUG Running `python scripts/publish/test_publish.py --uv ./uv all`
DEBUG uv 0.4.24 (4d2f[31](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:32)cb7 2024-10-21)
DEBUG Project `/home/runner/work/uv/uv` is marked as unmanaged
DEBUG Searching for default Python interpreter in managed installations or system path
DEBUG Found `cpython-3.12.7-linux-x86_64-gnu` at `/home/runner/.cache/uv/archive-v0/qzAMDq6g3xW3blrB8i8AY/bin/python3` (active virtual environment)
WARN Failed to get author from git: No author information found
DEBUG Writing Python versions to `/home/runner/work/uv/uv/scripts/publish/astral-test-token/.python-version`
DEBUG Detected existing version control system: git
Initialized project `astral-test-token` at `/home/runner/work/uv/uv/scripts/publish/astral-test-token`
Traceback (most recent call last):
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 272, in <module>

Publish astral-test-token for pypi-token
    main()
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 268, in main
    publish_project(project_name, uv)
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 1[38](https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925#step:7:39), in publish_project
    create_project(project_name, uv)
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 127, in create_project
    new_version = get_new_version(project_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 109, in get_new_version
    [_name, version] = parse_sdist_filename(filename)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/qzAMDq6g3xW3blrB8i8AY/lib/python3.12/site-packages/packaging/utils.py", line 154, in parse_sdist_filename
    raise InvalidSdistFilename(
packaging.utils.InvalidSdistFilename: Invalid sdist filename (extension must be '.tar.gz' or '.zip'): status page
DEBUG Command exited with code: 1
```

ref 

- https://github.com/astral-sh/uv/actions/runs/11430815307/job/31798960925
- https://github.com/astral-sh/uv/actions/runs/11430806304/job/31798947925

```
Run ./uv run -p 3.12 scripts/publish/test_publish.py --uv ./uv all
DEBUG uv 0.4.24 (159b664f9 2024-10-21)
Reading inline script metadata from `scripts/publish/test_publish.py`
DEBUG Searching for Python 3.12 in managed installations or system path
DEBUG Searching for managed installations at `/home/runner/.local/share/uv/python`
DEBUG Found `cpython-3.12.7-linux-x86_64-gnu` at `/opt/hostedtoolcache/Python/3.12.7/x64/bin/python` (search path)
DEBUG Caching via interpreter: `/opt/hostedtoolcache/Python/3.12.7/x64/bin/python`
DEBUG Using request timeout of 30s
DEBUG Solving with installed Python version: 3.12.7
DEBUG Solving with target Python version: >=3.12.7
DEBUG Adding direct dependency: httpx>=0.27.2, <0.28
DEBUG Adding direct dependency: packaging>=24.1, <25
DEBUG No cache entry for: https://pypi.org/simple/httpx/
DEBUG No cache entry for: https://pypi.org/simple/packaging/
DEBUG Searching for a compatible version of httpx (>=0.27.2, <0.28)
DEBUG Selecting: httpx==0.27.2 [compatible] (httpx-0.27.2-py3-none-any.whl)
DEBUG No cache entry for: https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb[23](https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134#step:7:24)3fe0416e4639146c089643407ce6/packaging-[24](https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134#step:7:25).1-py3-none-any.whl.metadata
DEBUG No cache entry for: https://files.pythonhosted.org/packages/56/95/9377bcb415797e44274b51d46e3249eba641711cf3348050f76ee7b15ffc/httpx-0.27.2-py3-none-any.whl.metadata
DEBUG Adding transitive dependency for httpx==0.27.2: anyio*
DEBUG Adding transitive dependency for httpx==0.27.2: certifi*
DEBUG Adding transitive dependency for httpx==0.27.2: httpcore>=1.dev0, <2.dev0
DEBUG Adding transitive dependency for httpx==0.27.2: idna*
DEBUG Adding transitive dependency for httpx==0.27.2: sniffio*
DEBUG Searching for a compatible version of packaging (>=24.1, <[25](https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134#step:7:26))
DEBUG Selecting: packaging==24.1 [compatible] (packaging-24.1-py3-none-any.whl)
DEBUG No cache entry for: https://pypi.org/simple/anyio/
DEBUG No cache entry for: https://pypi.org/simple/certifi/
DEBUG No cache entry for: https://pypi.org/simple/httpcore/
           ^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_api.py", line 118, in request
    return client.request(
           ^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_client.py", line 837, in request
    return self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_client.py", line 9[26](https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134#step:7:27), in send
    response = self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_client.py", line 954, in _send_handling_auth
    response = self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_client.py", line 991, in _send_handling_redirects
    response = self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_client.py", line 10[27](https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134#step:7:28), in _send_single_request
    response = transport.handle_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_transports/default.py", line 235, in handle_request
    with map_httpcore_exceptions():
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/home/runner/.cache/uv/archive-v0/N90949OWizxSQfDuicNRE/lib/python3.12/site-packages/httpx/_transports/default.py", line 89, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadTimeout: The read operation timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 272, in <module>
    main()
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 268, in main
    publish_project(project_name, uv)
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 1[38](https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134#step:7:39), in publish_project
    create_project(project_name, uv)
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 127, in create_project
    new_version = get_new_version(project_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 103, in get_new_version
    raise RuntimeError(f"Failed to fetch {url}") from err
RuntimeError: Failed to fetch https://codeberg.org/api/packages/astral-test-user/pypi/simple/astral-test-token
DEBUG Command exited with code: 1
```

ref 

- https://github.com/astral-sh/uv/actions/runs/11430785104/job/31798901134
- https://github.com/astral-sh/uv/actions/runs/11446764894/job/31846784598

```
DEBUG Detected existing version control system: git
Initialized project `astral-test-token` at `/home/runner/work/uv/uv/scripts/publish/astral-test-token`
Traceback (most recent call last):
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_transports/default.py", line 72, in map_httpcore_exceptions

Publish astral-test-token for pypi-token

Publish astral-test-password for pypi-password-env
    yield
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_transports/default.py", line 236, in handle_request
    resp = self._pool.handle_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/connection_pool.py", line 216, in handle_request
    raise exc from None
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/connection_pool.py", line 196, in handle_request
    response = connection.handle_request(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/connection.py", line 101, in handle_request
    return self._connection.handle_request(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/http11.py", line 143, in handle_request
    raise exc
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/http11.py", line 113, in handle_request
    ) = self._receive_response_headers(**kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/http11.py", line 186, in _receive_response_headers
    event = self._receive_event(timeout=timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_sync/http11.py", line 224, in _receive_event
    data = self._network_stream.read(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_backends/sync.py", line 124, in read
    with map_exceptions(exc_map):
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/contextlib.py", line 158, in __exit__

Publish astral-test-keyring for pypi-keyring

Publish astral-test-token for gitlab
    self.gen.throw(value)
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpcore/_exceptions.py", line 14, in map_exceptions
    raise to_exc(exc) from exc
httpcore.ReadTimeout: The read operation timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 101, in get_new_version
    data = client.get(url).text
           ^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_client.py", line 1066, in get
    return self.request(
           ^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_client.py", line 837, in request
    return self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_client.py", line 926, in send
    response = self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_client.py", line 954, in _send_handling_auth
    response = self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_client.py", line 991, in _send_handling_redirects
    response = self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_client.py", line 1027, in _send_single_request
    response = transport.handle_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_transports/default.py", line 235, in handle_request
    with map_httpcore_exceptions():
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/home/runner/.cache/uv/archive-v0/OIzkOzHQwO1eYhAmcTxfV/lib/python3.12/site-packages/httpx/_transports/default.py", line 89, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadTimeout: The read operation timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 273, in <module>
    main()
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 269, in main
    publish_project(project_name, uv, client)
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 138, in publish_project
    create_project(project_name, uv, client)
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 127, in create_project
    new_version = get_new_version(project_name, client)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/uv/uv/scripts/publish/test_publish.py", line 103, in get_new_version
    raise RuntimeError(f"Failed to fetch {url}") from err
RuntimeError: Failed to fetch https://codeberg.org/api/packages/astral-test-user/pypi/simple/astral-test-token
DEBUG Command exited with code: 1
Error: Process completed with exit code 1.
```

--- a/scripts/publish/test_publish.py
+++ b/scripts/publish/test_publish.py
@@ -61,7 +61,11 @@ from subprocess import PIPE, check_call, check_output, run
 from time import sleep
 
 import httpx
-from packaging.utils import parse_sdist_filename, parse_wheel_filename
+from packaging.utils import (
+    InvalidSdistFilename,
+    parse_sdist_filename,
+    parse_wheel_filename,
+)
 from packaging.version import Version
 
 TEST_PYPI_PUBLISH_URL = "https://test.pypi.org/legacy/"
@@ -128,14 +132,24 @@ def get_new_version(project_name: str, client: httpx.Client) -> Version:
     for target_config in all_targets.values():
         if target_config.project_name != project_name:
             continue
-        for filename in get_filenames(
-            target_config.index_url + project_name + "/", client
-        ):
-            if filename.endswith(".whl"):
-                [_name, version, _build, _tags] = parse_wheel_filename(filename)
+        url = target_config.index_url + project_name + "/"
+
+        # Get with retries
+        error = None
+        for _ in range(5):
+            try:
+                versions.update(collect_versions(url, client))
+            except httpx.HTTPError as err:
+                error = err
+                print(f"Error getting version, sleeping for 1s: {err}")
+                time.sleep(1)
+            except InvalidSdistFilename as err:
+                # Sometimes there's a link that says "status page"
+                error = err
+                print(f"Invalid index page, sleeping for 1s: {err}")
+                time.sleep(1)
             else:
-                [_name, version] = parse_sdist_filename(filename)
-            versions.add(version)
+                raise RuntimeError(f"Failed to fetch {url}") from error
     max_version = max(versions)
 
     # Bump the path version to obtain an empty version
@@ -144,21 +158,22 @@ def get_new_version(project_name: str, client: httpx.Client) -> Version:
     return Version(".".join(str(i) for i in release))
 
 
+def collect_versions(url: str, client: httpx.Client) -> set[Version]:
+    """Return all version from an index page."""
+    versions = set()
+    for filename in get_filenames(url, client):
+        if filename.endswith(".whl"):
+            [_name, version, _build, _tags] = parse_wheel_filename(filename)
+        else:
+            [_name, version] = parse_sdist_filename(filename)
+        versions.add(version)
+    return versions
+
+
 def get_filenames(url: str, client: httpx.Client) -> list[str]:
     """Get the filenames (source dists and wheels) from an index URL."""
-    # Get with retries
-    error = None
-    for _ in range(5):
-        try:
-            response = client.get(url)
-            data = response.text
-            break
-        except httpx.HTTPError as err:
-            error = err
-            print(f"Error getting version, sleeping for 1s: {err}")
-            time.sleep(1)
-    else:
-        raise RuntimeError(f"Failed to fetch {url}") from error
+    response = client.get(url)
+    data = response.text
     # Works for the indexes in the list
     href_text = r"<a(?: +[\w-]+=(?:'[^']+'|\"[^\"]+\"))* *>([^<>]+)</a>"
     return [m.group(1) for m in re.finditer(href_text, data)]

